src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -347,25 +347,28 @@ public FunctionDescriptor GetHttpFunctionOrNull(HttpRequestMessage request)
             if (routeData != null)
             {
                 _httpFunctions.TryGetValue(routeData.Route, out function);
+                AddRouteDataToRequest(routeData, request);
+            }
+
+            return function;
+        }
 
-                Dictionary<string, object> routeDataValues = null;
-                if (routeData.Values != null)
+        internal static void AddRouteDataToRequest(IHttpRouteData routeData, HttpRequestMessage request)
+        {
+            if (routeData.Values != null)
+            {
+                Dictionary<string, object> routeDataValues = new Dictionary<string, object>();
+                foreach (var pair in routeData.Values)
                 {
-                    routeDataValues = new Dictionary<string, object>();
-                    foreach (var pair in routeData.Values)
-                    {
-                        // filter out any unspecified optional parameters
-                        if (pair.Value != RouteParameter.Optional)
-                        {
-                            routeDataValues.Add(pair.Key, pair.Value);
-                        }
-                    }
+                    // translate any unspecified optional parameters to null values
+                    // unspecified values still need to be included as part of binding data
+                    // for correct binding to occur
+                    var value = pair.Value != RouteParameter.Optional ? pair.Value : null;
```

Comment:
This is the fix - previously we were filtering out optional unspecified params, meaning we weren't adding them to the binding contract in HttpTriggerBinding.